Makefile

```diff
@@ -225,6 +258,9 @@ _build/stage2/%: private GHC=$(realpath _build/stage1/bin/ghc)
 $(addprefix _build/stage2/bin/,$(STAGE2_EXECUTABLES)) &: $(CABAL) stage1
 	# Force cabal to replan
 	rm -rf _build/stage2/cache
+	$(call run_and_log, HADRIAN_SETTINGS='$(HADRIAN_SETTINGS)' \
+		PATH=$(PWD)/_build/stage1/bin:$(PATH) \
+		$(CABAL_BUILD) --ghc-options="-ghcversion-file=$(abspath ./rts/include/ghcversion.h)" -W $(GHC0) rts:nonthreaded-nodebug )
```

Comment:
This really feels like a bug, why do we need to build this explicitly here, why is this not brought into scope automatically?